components/bl-markdown-component/src/index.js

```diff
@@ -0,0 +1,85 @@
+import { useEffect, useMemo, useState } from 'react';
+
+import MarkdownIt from './markdown-it.min.js';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+export default function MyCustomComponent({ component }) {
+  const { classList, style, display, url, markdownText, height, width } = component;
+
+  const [markdown, setMarkdown] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const mdUrl = useMemo(() => url, [url]);
+
+  const md = useMemo(() => new MarkdownIt(), []);
+
+  component.setUrl = (url) => {
+    loadMd(url);
+  };
+
+  component.setContent = (text) => {
+    loadMd(mdUrl, text);
+  };
+
+  useEffect(() => {
+    loadMd(mdUrl, markdownText);
+  }, [markdownText, mdUrl]);
+
+  const loadMd = (url, text) => {
```

Comment:
Could we split it into two functions setContent/fetchContent or something like that?